src/generator/ArtifactGenerator.js

```diff
@@ -171,105 +171,113 @@ class ArtifactGenerator {
-            vocabDetails.nameAndPrefixOverride;
-          this.artifactData.namespaceOverride = vocabDetails.namespaceOverride;
-
-          this.artifactData.ignoreNonVocabTerms =
-            vocabDetails.ignoreNonVocabTerms;
-
-          // Just in case the vocabulary itself does not provide a description
-          // of itself, we pass down the description from our configuration as
-          // a fallback (and prefix that description with some text to denote
-          // that it's not coming from the original vocabulary itself).
-          this.artifactData.descriptionFallback = `[Generator provided] - ${vocabDetails.description}`;
-
-          // Generate this vocab for each artifact we are generating for.
-          const artifactPromises = this.artifactData.artifactToGenerate.map(
-            (artifactDetails) => {
-              return new VocabGenerator(this.artifactData, artifactDetails)
-                .generate()
-                .catch((error) => {
-                  const message = `Failed generation: [${error.message}]`;
-                  debug(message);
-                  throw new Error(message);
-                });
-            }
-          );
+      this.artifactData.vocabContentTypeHeaderOverride =
+        vocabDetails.vocabContentTypeHeaderOverride;
 
-          // Wait for all our artifacts to be generated.
-          await Promise.all(artifactPromises);
+      this.artifactData.vocabContentTypeHeaderFallback =
+        vocabDetails.vocabContentTypeHeaderFallback;
 
-          // Only return the first one, as we don't want duplicate info.
-          return artifactPromises[0];
-        })
-      );
+      this.artifactData.nameAndPrefixOverride =
+        vocabDetails.nameAndPrefixOverride;
+      this.artifactData.namespaceOverride = vocabDetails.namespaceOverride;
+
+      this.artifactData.ignoreNonVocabTerms = vocabDetails.ignoreNonVocabTerms;
+
+      // Just in case the vocabulary itself does not provide a description
+      // of itself, we pass down the description from our configuration as
+      // a fallback (and prefix that description with some text to denote
+      // that it's not coming from the original vocabulary itself).
+      this.artifactData.descriptionFallback = `[Generator provided] - ${vocabDetails.description}`;
+
+      const generatedArtifacts = [];
+      // Generate this vocab for each artifact we are generating for.
+      for (let i = 0; i < this.artifactData.artifactToGenerate.length; i++) {
+        const artifactDetails = this.artifactData.artifactToGenerate[i];
+
+        try {
+          generatedArtifacts.push(
+            await new VocabGenerator(
+              this.artifactData,
+              artifactDetails
+            ).generateVocab()
+          );
+        } catch (error) {
+          const message = `Failed generation: [${error.message}]`;
+          debug(message);
+          throw new Error(message);
+        }
+      }
```

Comment:
Ditto regarding await in loop and https://www.npmjs.com/package/p-limit

Reply:
Yeah, and ditto my answer back above too I think :) !